vsock/src/txbuf.rs

```diff
@@ -0,0 +1,210 @@
+use super::vhu_vsock::{Error, Result, CONN_TX_BUF_SIZE};
+use std::{io::Write, num::Wrapping};
+
+#[derive(Debug)]
+pub struct LocalTxBuf {
+    /// Buffer holding data to be forwarded to a host-side application
+    buf: Vec<u8>,
+    /// Index into buffer from which data can be consumed from the buffer
+    head: Wrapping<u32>,
+    /// Index into buffer from which data can be added to the buffer
+    tail: Wrapping<u32>,
+}
+
+impl LocalTxBuf {
```

Comment:
I've just been looking at this code and wondering why we can't use an existing buffer implementation from the standard crates?

Reply:
Does the standard library provide such crate? We are usually looking to reduce the number of external dependencies. I will continue working on upstreaming this and the other vsock components sometime in the near future, but if we can't use something from `std`, I think I would be in favor of using a local implementation (hopefully we can simplify the one we have here).